File: common/src/main/java/guepardoapps/lucahome/common/controller/deprecated/ICommandController.java
package guepardoapps.lucahome.common.controller.deprecated;

import android.support.annotation.NonNull;

public interface ICommandController {
    boolean RebootDevice(int timeout);

    boolean ShutDownDevice(int timeout);

    boolean HasRoot();

    boolean PerformCommand(@NonNull String[] command);
}

File: common/src/main/java/guepardoapps/lucahome/common/controller/ICommandController.kt
package guepardoapps.lucahome.common.controller

interface ICommandController {
    fun rebootDevice(timeoutMs: Long): Boolean
    fun shutDownDevice(timeoutMs: Long): Boolean

    @Throws(InterruptedException::class)
    fun simulateKeyPress(keys: IntArray, timeout: Int)
}